### PR TITLE
feat: chat functionality with emojis added

### PR DESCRIPTION
**Pull request by PTRG01**

### CODE REVIEW

File: apps/client/src/app/components/chat/chat.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    },
+  });
+
+  const insertAtCursor = (message: string, emoji: string) => {
+    const input = emoji.trim();
+
+    if (!input) {
+      return message;
+    }
+    const textarea = document.getElementById('textarea');
```
Comment (AdamGulmantowicz) [inline apps/client/src/app/components/chat/chat.tsx:115]: This is not the way how to use reference to DOM element. Please see `useRef` hook in react docs.

File: apps/client/src/app/components/chats/chats.module.css [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.chat-container {
+  position: fixed;
+  bottom: 4rem;
```
Comment (AdamGulmantowicz) [inline apps/client/src/app/components/chats/chats.module.css:3]: Why it's 4rem?

File: apps/client/src/app/components/chats/chats.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import { useChatContext } from '../../contexts/chat-provider/chat-provider';
+import Chat from '../chat/chat';
+import styles from './chats.module.css';
+/* eslint-disable-next-line */
+export interface IChatsProps {}
+
+export function Chats(props: IChatsProps) {
+  const { openChats } = useChatContext();
+  return (
+    <div className={styles['chat-container']}>
```
Comment (AdamGulmantowicz) [inline apps/client/src/app/components/chats/chats.tsx:11]: What if there are no chats?

File: apps/client/src/app/components/following-list/following-list.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          ? data.expand.followers.map((item: IUser) => (
+              <UserListItem
+                label={item.name}
+                avatar={item.avatar}
+                id={item.id}
+                key={item.id}
+                values={user?.followers}
+                onAddValue={function (): void {
+                  throw new Error('Function not implemented.');
+                }}
```
Comment (AdamGulmantowicz) [inline apps/client/src/app/components/following-list/following-list.tsx:47]: Why it's typed void?

File: apps/client/src/app/components/test-component/test-component.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import styles from './test-component.module.css';
+
+/* eslint-disable-next-line */
+export interface TestComponentProps {}
+
+export function TestComponent(props: TestComponentProps) {
+  return (
+    <div className={styles['container']}>
+      <h1>Welcome to TestComponent!</h1>
```
Comment (AdamGulmantowicz) [inline apps/client/src/app/components/test-component/test-component.tsx:9]: I think it's unnecessary.

**Review by AdamGulmantowicz — CHANGES_REQUESTED** [review]
It could be good to remove leftovers, wrap functions in `useCallback`, add translations and also move functions from js to separate declarations to increase redability.

File: apps/client/src/app/components/chats/chats.tsx (hunk 3 above)
Comment (PTRG01) [reply apps/client/src/app/components/chats/chats.tsx:11]: no chats message added to chats-list - chats component serves only as display container for chat windows

File: apps/client/src/app/components/message/message.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  avatar?: string;
+  id: string;
+  content: string;
+  timestamp?: string;
+}
+
+export function Message(props: MessageProps) {
+  const { user } = useAuthContext();
+  const dateObj = new Date(props.timestamp);
+  const formattedDate = `${dateObj.toLocaleString('en-US', {
```
Comment (AdamGulmantowicz) [inline apps/client/src/app/components/message/message.tsx:15]: You could `useMemo`

**Pull request merged**